superset/advanced_data_type/plugins/agent_id.py

```diff
@@ -0,0 +1,47 @@
+import re
+from superset.advanced_data_type.plugins import translate_filter_func
+from superset.advanced_data_type.plugins.operator_sets import EQUAL_NULLABLE_OPERATOR_SET, PATTERN_MATCHING_OPERATOR_SET
+from superset.advanced_data_type.types import AdvancedDataType, AdvancedDataTypeRequest, AdvancedDataTypeResponse
+
+
+
+def agent_id_func(req: AdvancedDataTypeRequest) -> AdvancedDataTypeResponse:
+    """
+    Convert a passed in AdvancedDataTypeRequest to an AdvancedDataTypeResponse
+    """
+    resp: AdvancedDataTypeResponse = {
+        "values": [],
+        "error_message": "",
+        "display_value": "",
+        "valid_filter_operators": EQUAL_NULLABLE_OPERATOR_SET + PATTERN_MATCHING_OPERATOR_SET,
+    }
+
+    if req["operator"] in ['IS NULL', 'IS NOT NULL']:
+        return resp
+    elif req["values"] == [""]:
+        resp["error_message"] = "Agent ID must not be empty"
+        return resp
+    elif req["operator"] in ['LIKE', 'ILIKE']:
+        for val in req["values"]:
+            resp["values"].append(str(val))
+        return resp
+    else:
+        for val in req["values"]:
+            string_value = str(val)
+            if re.search("^[a-zA-Z0-9]{2}.[a-zA-Z0-9]{2}.[a-zA-Z0-9]{2,}.[a-zA-Z0-9]{2}$", string_value):
+                resp["values"].append(string_value)
+            else:
+                resp["error_message"] = f"'{ val }' is not a valid Agent ID. Must be four strings separated by periods, and of lengths 2.2.*.2"
```

Comment:
The stated format for Agent ID is wrong. Here is what I got from Jim Don (we should follow up to find out more about those special cases):

> The format is roughly ...
> ffff.ff.ffffffff.ff
> 64 bits ... left to right:
> 16 for client id (ffff)
> 8 for subnet (ff)
> 32 for host unique id (ffffffff)
> 8 for platform and os (ff)
> There's also magic values like 1 and f that mean things ... but the above is the basic format.

Reply:
I've also confirmed with Jim that the example format above represents the max length of each entry in the tuple. So that first entry is `ffff`, but it could be 1-4 hex digits in length. The second is 1-2 and the third is 1-8. The fourth entry is the only exception. Each nibble has its own meaning, so it will always be 2 hex digits in length.

Reply:
I also got clarification about those "magic values", but in the end, those only matter when interpreting what each entry means. Format-wise those special values change nothing, so they have no impact on our validation code here.